### PR TITLE
Split GivenThatWeWantToRunILLink test class into multiple so Helix can run them separately

### LINKED ISSUE #44895: Microsoft.NET.Publish.Tests.dll.5 is timing out a lot, imbalance with other partitions
This workitem is one of the most frequent to time out: https://github.com/dotnet/sdk/issues/40074

A quick Kusto query shows that we're already pretty close to the default 45mins timeout for a lot of _passing_ runs:

image

This is Windows runs in the last 14 days with the x axis being the execution time. Note that the bucket on the left are runs which had test failures which still count as Status==Pass in Helix terms.

The other partitions of Microsoft.NET.Publish.Tests.dll are all in the 10-15mins range so we probably got unlucky and the long running tests are all in the .5 partition.

Bumping the timeout would be an option or increasing the number of partitions, are there other things we could do?

/cc @marcpopMSFT @baronfel

### PR DESCRIPTION
**Pull request by akoeplinger**
Fixes https://github.com/dotnet/sdk/issues/44895

Durations from this PR:

Durations from a random other PR (#44968):

### CODE REVIEW

File: test/Microsoft.NET.Publish.Tests/GivenThatWeWantToRunILLink.cs [hunk 1]
```diff
@@ -1811,8 +1774,13 @@ public void Build_respects_PublishTrimmed_property(string targetFramework)
             // just setting PublishTrimmed doesn't inject the IsTrimmable attribute
             AssemblyInfo.Get(assemblyPath).ContainsKey("AssemblyMetadataAttribute").Should().BeFalse();
         }
+    }
+
+    internal static class ILLinkTestUtils
+    {
```
Comment (akoeplinger) [inline test/Microsoft.NET.Publish.Tests/GivenThatWeWantToRunILLink.cs:1780]: I kept the utility functions in the same file to minimize the diff but let me know if I should move it to a separate file

**Review by jtschuster — APPROVED** [review]
LGTM, thank you!

**Review by sbomer — APPROVED** [review]
Nice!

**Pull request merged**